### PR TITLE
Reset all PGA volume to 0dB

### PR DESCRIPTION
**Pull request by keqiaozhang**

### CODE REVIEW

**Comment by marc-hb** [comment]
Can you clarify the relationship with
- #951?

**Comment by keqiaozhang** [comment]
> Can you clarify the relationship with
> 
> * [alsa_settings: CAVS_NOCODEC: reset values to 0dB/on with script #951](https://github.com/thesofproject/sof-test/pull/951)?

I think https://github.com/thesofproject/sof-test/pull/951 is only reset the volume for nocodec devices. But we found the plain arecord issue(https://github.com/thesofproject/linux/issues/3804) on many other models recently:
http://sof-ci.sh.intel.com/#/result/planresultdetail/14968?model=ADLP_SKU0B00_SDCA_ZEPHYR&testcase=multiple-pause-resume-50
http://sof-ci.sh.intel.com/#/result/planresultdetail/14926?model=ADLP_BRYA_I2S&testcase=check-pause-resume-capture-100
...
This is because the PGA capture volume being turned to the maximum after the `volume_basic_test.sh`. So I added a function to reset the PGA volume to 0dB for all our models at the end of the volume test.

File: case-lib/lib.sh [hunk 1]
```diff
@@ -768,4 +768,14 @@ set_alsa_settings()
     esac
 }
 
+reset_PGA_volume()
+{
+    # set all PGA* volume to 0dB
+    amixer -Dhw:0 scontrols | sed -e "s/^.*'\(.*\)'.*/\1/" |grep PGA | \
```
Comment (marc-hb) [inline case-lib/lib.sh:774]: You don't need the `\` backslash, ending a line with `|  ` is enough

Comment (marc-hb) [reply case-lib/lib.sh:774]:
```suggestion
    amixer -Dhw:0 scontrols | sed -e "s/^.* '\(.*\)',.*/\1/" |grep PGA | 
```

File: test-case/volume-basic-test.sh [hunk 2]
```diff
@@ -89,4 +89,7 @@ done
 #clean up background aplay
 pkill -9 aplay
 
+dlogi "Reset all PGA volume to 0dB"
+reset_PGA_volume || die "Failed to reset the PGA volume to 0dB."
```
Comment (marc-hb) [inline test-case/volume-basic-test.sh:93]:
```suggestion
reset_PGA_volume || die "Failed to reset some PGA volume to 0dB."
```

File: test-case/volume-basic-test.sh [hunk 3]
```diff
@@ -89,4 +89,7 @@ done
...
+
```
Comment (marc-hb) [inline test-case/volume-basic-test.sh:94]: I know it's off-topic but please remove this spurious `exit 0` if you ever force-push a new version.

**Review by marc-hb — COMMENTED** [review]
What do you think of this warning? I know it's off-topic but it looks like a serious bug; serious enough to require fixing before changing other, more minor problems.

```
In test-case/volume-basic-test.sh line 65:
for i in $(seq 1 $maxloop)
^-^ SC2167: This parent loop has its index variable overridden.


In test-case/volume-basic-test.sh line 70:
    for i in "${pgalist[@]}"
    ^-^ SC2165: This nested loop overrides the index variable of its parent.
```

File: test-case/volume-basic-test.sh (hunk 2 above)
Comment (keqiaozhang) [reply test-case/volume-basic-test.sh]: Fixed.

File: case-lib/lib.sh (hunk 1 above)
Comment (keqiaozhang) [reply case-lib/lib.sh]: Updated.

**Comment by keqiaozhang** [comment]
> serious enough to require fixing before changing other, more minor problems.

I will fix these warnings in another PR next.

**Review by fredoh9 — DISMISSED** [review]
Thank you for finding the root cause why all PGAs are 100%, I totally forgot about volume test. Resetting to 0 db works.
And shellcheck fixes are also good catch.

Let volume test run first before alsabat test after this.

**Comment by marc-hb** [comment]
> replace the index name to kctl for the nested loop
> it may overrides the index variable of its parent.

Thanks for the fix. I haven't read the code but it looks like a very serious fix? Maybe the title of the commit message should not be "fix shellcheck warning" like it's just pleasing shellcheck and nothing else. The reason why we use shellcheck is because sometimes it can find serious bugs like this one and it's worth all the other times when it reports non-serious issues.

**Comment by keqiaozhang** [comment]
> Let volume test run first before alsabat test after this.

I will change the test case order.

**Comment by keqiaozhang** [comment]
> Maybe the title of the commit message should not be "fix shellcheck warning" like it's just pleasing shellcheck and nothing else.

I have revised the commit message.

File: case-lib/lib.sh [hunk 4]
```diff
@@ -768,4 +768,14 @@ set_alsa_settings()
     esac
 }
 
+reset_PGA_volume()
+{
+    # set all PGA* volume to 0dB
+    amixer -Dhw:0 scontrols | sed -e "s/^.*'\(.*\)'.*/\1/" |grep PGA |
```
Comment (aiChaoSONG) [inline case-lib/lib.sh:774]: for IPC4 the volume control are named as gain, I think you should also grep gain here.

Comment (keqiaozhang) [reply case-lib/lib.sh:774]: Yes, you are right. But the volume-base-test.sh only support PGA control ATM. The volume test is skipped on all our IPC4 platforms. I can submit another PR for IPC4.

**Review by aiChaoSONG — APPROVED** [review]

**Comment by aiChaoSONG** [comment]
merge for v2.3 test.

**Pull request merged**